test_delivery.go

```diff
@@ -4,17 +4,17 @@ import "encoding/json"
 
 //go:generate stringer -type=state
 
-type state int
+type State int
```

Comment:
I think we should move the now public `State` struct to its own file `state.go` now.